test/client-pub-sub.js

```diff
@@ -395,3 +395,43 @@ test('get message when client disconnects', function (t) {
     t.equal(client2, packet.payload.toString())
   })
 })
+
+test('should not receive a message on negated subscription', function (t) {
+  t.plan(3)
+
+  var broker = aedes()
+  broker.authorizeSubscribe = function (client, sub, callback) {
+    callback(null, null)
+  }
+
+  broker.on('client', function (client) {
+    broker.publish({
+      topic: 'hello',
+      payload: Buffer.from('world'),
+      qos: 0,
+      retain: true
+    }, function (err) {
+      t.error(err, 'no error')
+      client.subscribe({
+        topic: 'hello',
+        qos: 0
+      }, function (err) {
+        t.error(err, 'no error')
+      })
+    })
+  })
+
+  var s = connect(setup(broker))
+  var receivedPacket = null
+  s.outStream.once('data', function (packet) {
+    receivedPacket = packet
```

Comment:
I think you can just call `t.fail()` here.

Reply:
The tests process would not exit using this approach as `t.fail()` will never be called and the plan takes 3 assertions.